woocommerce/woocommerce-gutenberg-products-block
# Collection Filters refactor: data fetching, context passing, and code organization

## What

After examining the work in woocommerce/woocommerce-blocks#11191, we (@dinhtungdu and @samueljseay) concludes that's not the good direction to move forward. This PR collects good implementation from woocommerce/woocommerce-blocks#11191 like context passing and code organization to update the current code base. This PR also fixes the collection data fetching issue for the Collection Filters block.

Fixes woocommerce/woocommerce-blocks#10813, replaces woocommerce/woocommerce-blocks#11191 

## Why

- woocommerce/woocommerce-blocks#11191 makes the `Collection Filters` block really bloated.
- We wanted to make filter blocks only consume the data passed from `Collection Filters`, not fetching data. But this is impossible, filter blocks like Attribute Filter needs to fetch its own data => this make developers confused by having data fetching logic in two places.



## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

Ensure testing step listed in woocommerce/woocommerce-blocks#10730 still apply.


## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


> Add suggested changelog entry here.
